mlflow/server/handlers.py

```diff
@@ -2270,9 +2458,22 @@ def _get_paths(base_path):
     We should register paths like /api/2.0/mlflow/experiment and
     /ajax-api/2.0/mlflow/experiment in the Flask router.
     """
+    base_path = _convert_path_parameter_to_flask_format(base_path)
     return [_get_rest_path(base_path), _get_ajax_path(base_path)]
 
 
+def _convert_path_parameter_to_flask_format(path):
+    """
+    Converts path parameter format to Flask compatible format.
+
+    Some protobuf endpoint paths contain parameters like /mlflow/trace/{request_id}.
+    This can be interpreted correctly by gRPC framework like Armeria, but Flask does
+    not understand it. Instead, we need to specify it with a different format,
+    like /mlflow/trace/<request_id>.
+    """
+    return re.sub(r"{(\w+)}", r"<\1>", path)
```

Comment:
We can update `service.proto` if that works.

Reply:
We use `<...>` here in `mlflow_artifacts.proto`:

https://github.com/mlflow/mlflow/blob/2ac15ae4eb9da0d6653a5b7129578088cf32e30a/mlflow/protos/mlflow_artifacts.proto#L62

Reply:
Not a blocker.

Reply:
@harupy I was not sure if it works with Armeria in Databricks side: none of endpoint has the `<>` format https://github.com/databricks/universe/tree/master/mlflow/api/proto.

Reply:
We don't necessarily have to. Manual adjustments for OSS are acceptable.

Reply:
Okie cool, then let me simple use the curly bracket in OSS🙂

Reply:
Sounds good. I think `<>` will break the following line:

https://github.com/mlflow/mlflow/blob/72b256221e48a81e0d0a790f8886b6527756cb13/mlflow/store/artifact/databricks_artifact_repo.py#L174

We can use the same approach as `RestStore._call_endpoint`:

https://github.com/mlflow/mlflow/blob/72b256221e48a81e0d0a790f8886b6527756cb13/mlflow/store/tracking/rest_store.py#L74-L79

Reply:
Oh I see, in that case, keeping the curly bracket and formatting for Flask is actually simpler? It seems Flask is the outsider that cannot understand the format😅